pymtl3/passes/sim/AddSimUtilFuncsPass.py

```diff
@@ -0,0 +1,50 @@
+"""
+========================================================================
+AddSimUtilFuncsPass.py
+========================================================================
+
+Author : Shunning Jiang
+Date   : Jan 26, 2020
+"""
+
+from pymtl3.datatypes import b1
+from pymtl3.passes.BasePass import BasePass, PassMetadata
+from pymtl3.passes.errors import PassOrderError
+
+
+class AddSimUtilFuncsPass( BasePass ):
+
+  def __call__( self, top ):
+    if not hasattr( top, "tick" ):
+      raise PassOrderError( "tick" )
+
+    if hasattr(top, "sim_reset"):
+      raise AttributeError( "Please modify the attribute top.sim_reset to "
+                            "a different name.")
+    if hasattr(top, "print_line_trace"):
+      raise AttributeError( "Please modify the attribute top.print_line_trace to "
+                            "a different name.")
+
+    top.sim_reset = self.create_reset( top, reset_value=1 )
+    top.print_line_trace = self.create_print_line_trace( top )
+
+  @staticmethod
+  # Simulation related APIs
+  def create_reset( top, reset_value ):
+    assert reset_value in [0,1]
+    def reset(print_line_trace=False):
+      print()
```

Comment:
`reset_value` doesn't sound right... It's like the default value that gets loaded into register when reset is enablesd. Instead I think it should be `active_high=True`. In this way we can get rid of that assertion as well.

Some unimportant stuff: empty space after `(` and before `)` and we probably can remove this print?

Reply:
done